edxval/models.py

```diff
@@ -155,6 +170,84 @@ class EncodedVideo(models.Model):
     video = models.ForeignKey(Video, related_name="encoded_videos")
 
 
+class CustomizableImageField(models.ImageField):
+    """
+    Subclass of ImageField that allows custom settings to not
+    be serialized (hard-coded) in migrations. Otherwise,
+    migrations include optional settings for storage (such as
+    the storage class and bucket name); we don't want to
+    create new migration files for each configuration change.
+    """
+    def __init__(self, *args, **kwargs):
+        kwargs.update(dict(
+            upload_to=video_image_path,
+            storage=get_video_image_storage(),
+            max_length=500,  # allocate enough for filepath
+            blank=True,
+            null=True
+        ))
+        super(CustomizableImageField, self).__init__(*args, **kwargs)
+
+    def deconstruct(self):
+        """
+        Override base class method.
+        """
+        name, path, args, kwargs = super(CustomizableImageField, self).deconstruct()
+        del kwargs['upload_to']
+        del kwargs['storage']
+        del kwargs['max_length']
+        return name, path, args, kwargs
+
+
+class VideoImage(TimeStampedModel):
+    """
+    Image model for course video.
+    """
+    course_video = models.OneToOneField(CourseVideo, related_name="video_image")
+    image = CustomizableImageField()
+
+    @classmethod
+    def create_or_update(cls, course_video, file_name, image_data=None):
+        """
+        Create a VideoImage object for a CourseVideo.
+
+        Arguments:
+            course_video (CourseVideo): CourseVideo instance
+            file_name (str): File name of the image
```

Comment:
I think in few cases this will be user-specified name and in some cases this will be existing name, can we make it clear in doc strings ?